atlasdb-client/src/main/java/com/palantir/atlasdb/ptobject/EncodingUtils.java

```diff
@@ -464,4 +464,21 @@ public static Long decodeFlippedNullableFixedLong(byte[] value, int offset) {
             return ArrayUtils.addAll(new byte[] { 1 }, PtBytes.toBytes(Long.MIN_VALUE ^ value));
         }
     }
+
+    public static byte[] reverseBits(byte[] value) {
+        ArrayUtils.reverse(value);
+        for (int index = 0; index < value.length; index++) {
+            value[index] = reverseBitsInByte(value[index]);
+        }
+        return value;
+    }
+
+    private static byte reverseBitsInByte(byte value) {
+        byte result = 0;
+        for (int index = Byte.SIZE - 1; index >= 0; index--) {
```

Comment:
these methods often turn out to be weirdly slow (if you load 1k timestamps out of Cassandra, you'll need to spend legit a few million timestamps (1000 * probably 4 cycles per loop iteration * Long.SIZE)) just decoding.

Reply:
ok so that math was clearly very blatantly wrong

Reply:
but it's still millions of cycles a second on some of our larger stacks. Basically, in cases like this it's worth making a `byte[]` which just does the mapping for you for all byte values.

Reply:
Ah, yeah a lookup table makes sense since there are only 256 possibilities - though I agree on `Long.reverse`, I was looking for a library method to do this stuff but had no idea that was a thing